runtime/tests/Test_NullQubit.cpp

```diff
@@ -598,3 +598,106 @@ TEST_CASE("Test NullQubit device shots methods", "[NullQubit]")
         CHECK(sim->GetDeviceShots() == i);
     }
 }
+
+TEST_CASE("Test NullQubit device resource tracking", "[NullQubit]")
+{
+    // The name of the file where the resource usage data is stored
+    constexpr char RESOURCES_FNAME[] = "__pennylane_resources_data.json";
+
+    // Open a file for writing the resources JSON
+    std::ofstream resource_file_w(RESOURCES_FNAME);
+    if (!resource_file_w.is_open()) { // LCOV_EXCL_LINE
+        FAIL("Failed to open resource usage file for writing.");
```

Comment:
Do you need to add `// LCOV_EXCL_LINE` to L610 to fix the codecov warning?

Reply:
Fixed in [fbf4a94](https://github.com/PennyLaneAI/catalyst/pull/1619/commits/fbf4a94665b6d09f76b767c969bc6b17100da3d3)